SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/OwnershipLockingPolicy.cpp

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LoadBalancing/OwnershipLockingPolicy.h"
+
+#include "EngineClasses/AbstractSpatialPackageMapClient.h"
+#include "Interop/SpatialStaticComponentView.h"
+#include "Schema/AuthorityIntent.h"
+#include "Schema/Component.h"
+#include "Utils/SpatialActorUtils.h"
+
+#include "Improbable/SpatialEngineDelegates.h"
+#include "UObject/UObjectGlobals.h"
+
+DEFINE_LOG_CATEGORY(LogOwnershipLockingPolicy);
+
+bool UOwnershipLockingPolicy::CanAcquireLock(const AActor* Actor) const
+{
+	if (Actor == nullptr)
+	{
+		UE_LOG(LogOwnershipLockingPolicy, Error, TEXT("Failed to lock nullptr actor"));
+		return false;
+	}
+
+	check(PackageMap.IsValid());
+	Worker_EntityId EntityId = PackageMap.Get()->GetEntityIdFromObject(Actor);
+	if (EntityId == SpatialConstants::INVALID_ENTITY_ID)
+	{
+		UE_LOG(LogOwnershipLockingPolicy, Error, TEXT("Failed to lock actor without corresponding entity ID. Actor: %s"), *Actor->GetName());
+		return false;
+	}
+
+	check(StaticComponentView.IsValid());
+	const bool bHasAuthority = StaticComponentView.Get()->HasAuthority(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID);
+	if (!bHasAuthority)
+	{
+		UE_LOG(LogOwnershipLockingPolicy, Warning, TEXT("Can not lock actor migration. Do not have authority. Actor: %s"), *Actor->GetName());
+		return false;
+	}
+
+	check(VirtualWorkerTranslator != nullptr);
+	const bool bHasAuthorityIntent = VirtualWorkerTranslator->GetLocalVirtualWorkerId() ==
+		StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(EntityId)->VirtualWorkerId;
+	if (!bHasAuthorityIntent)
+	{
+		UE_LOG(LogOwnershipLockingPolicy, Warning, TEXT("Can not lock actor migration. Authority intent does not match this worker. Actor: %s"), *Actor->GetName());
```

Comment:
These seems like an error, not a warning.

Reply:
This was outside of PR changes, just file rename. Could change if you think but I don't have strong opinions either way. If this warning is hit, you're AcquireLock will return an invalid lock token which you can guard around. Warning might be fine.